eng/scripts/Set-CliVersionInBuild.ps1

```diff
@@ -0,0 +1,33 @@
+<#
+    .SYNOPSIS
+    Set-CliVersionInBuild sets the CLI version defined in version.txt.
+
+    .PARAMETER BuildReason
+    The build reason supplied from the CI provider. In Azure Pipelines, this would be "Build.Reason".
+
+    .PARAMETER BuildId
+    A unique build ID supplied from the CI provider. In Azure Pipelines, this would be "Build.BuildId".
+ #>
+param(
+    [string]$BuildReason,
+    [string]$BuildId
+)
+
+$prereleaseCategory = ""
+
+if ($BuildReason -eq "Manual") {
+    Write-Host "Skipping prerelease tagging for release build."
+    exit 0
+}
+elseif ($BuildReason -eq "PullRequest") {
+    $prereleaseCategory = "pr"
+}
+else {
```

Comment:
This works because other values for `$BuildReason` amount to a "daily" build (i.e. `Schedule`)... We haven't used `ResourceTrigger` or other build types at this point. I think this is a reasonable default for now.

Reply:
Add comment to clarify this.